src/pages/api/import-exhibitor.ts

```diff
@@ -0,0 +1,128 @@
+import type { NextApiRequest, NextApiResponse } from "next";
+import { prisma } from "@/server/db";
+import { env } from "@/env.mjs";
+import { timingSafeEqual, randomUUID } from "crypto";
+import { validateOrganizationNumber } from "@/shared/validateOrganizationNumber";
+
+const importToken = Buffer.from(env.IMPORT_TOKEN);
+
+export default async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse
+) {
+  const authHeader = req.headers["authorization"];
+  if (req.method !== "PUT") return res.status(405).end();
+  else if (
+    authHeader == undefined ||
+    authHeader.length != importToken.length ||
+    !timingSafeEqual(Buffer.from(authHeader), importToken)
+  ) {
+    return res.status(402).end();
+  }
+
+  const {
+    contactPerson,
+    telephoneNumber,
+    companyName,
+    organizationNumber,
+    email,
+    exhibitorPackage,
+  } = req.body;
+
+  const v = validateOrganizationNumber(
+    typeof organizationNumber === "string" ? organizationNumber : ""
+  );
+
+  if (
```

Comment:
I think tt would be nicer to use zod here. Something like:
```ts
const bodySchema = z.object({
  contactPerson: z.string(),
  telephoneNumber: z.string(),
  companyName: z.string(),
  organizationNumber: z.string().transform((s, ctx) => {
    const orgNum = validateOrganizationNumber(s);
    if ("error" in orgNum) {
      ctx.addIssue({
        code: z.ZodIssueCode.custom,
        message: orgNum.error,
      });
      return z.NEVER;
    }
    return orgNum.value;
  }),
  email: z.string(),
  exhibitorPackage: z.enum(["base", "sponsor", "headhunter", "premium"]),
});

const body = bodySchema.safeParse(req.body);
if (!body.success) {
  return res.status(400).end(body.error);
}

const {
  contactPerson,
  telephoneNumber,
  companyName,
  organizationNumber,
  email,
  exhibitorPackage,
} = body.data;
```
Then you don't have to cast the package later.

This makes me realize that it would probably be nice to have a zod-wrapper for the organization number validation

Reply:
Nice solution!